src/code/audioMgr.c

```diff
@@ -0,0 +1,137 @@
+#include "global.h"
+
+void func_80172C30(AudioMgr* audioMgr) {
+    AudioTask* task = audioMgr->rspTask;
+
+    if (audioMgr->rspTask->taskQueue != NULL) {
+        osSendMesg(task->taskQueue, NULL, OS_MESG_BLOCK);
+    }
+}
+
+void AudioMgr_HandleRetrace(AudioMgr* audioMgr) {
+    static s32 D_801D14F0 = 10;
+    AudioTask* rspTask;
+    s32 timerMsgVal = 666;
+    OSTimer timer;
+    s32 msg;
+
+    if (SREG(20) > 0) {
+        audioMgr->rspTask = NULL;
+    }
+
+    while (audioMgr->unk_F4.validCount != 0) {
+        osRecvMesg(&audioMgr->unk_F4, NULL, OS_MESG_NOBLOCK);
+    }
+
+    if (audioMgr->rspTask != NULL) {
+        audioMgr->audioTask.next = NULL;
+        audioMgr->audioTask.flags = 2;
```

Comment:
```suggestion
        audioMgr->audioTask.flags = OS_SC_NEEDS_RSP;
```
Not sure if this constant is in any mm headers yet, if not get it from oot.